### ISSUE TITLE
@NotNull should apply to DDL

### ISSUE THREAD

**Issue by jo-elimu** [elimu-ai/webapp#1702]
The JPA schema export at https://github.com/elimu-ai/webapp/blob/main/src/main/resources/META-INF/jpa-schema-export.sql should apply `not null` when an entity's property is annotated with `@NotNull`.

https://stackoverflow.com/questions/51258073/notnull-annotation-does-not-translate-to-not-null-constraint-in-schema-generati

---

Relates to #1286, #1607

**Comment by venkatesh2k3** [comment]
@jo-elimu So, do we need to add null constraints to the model classes according to this file [](https://github.com/elimu-ai/webapp/blob/main/src/main/resources/META-INF/jpa-schema-export.sql)

**Comment by jo-elimu** [comment]
> @jo-elimu So, do we need to add null constraints to the model classes according to this file https://github.com/elimu-ai/webapp/blob/main/src/main/resources/META-INF/jpa-schema-export.sql

@venkatesh2k3 No, we want `not null` to be applied _automatically_ to the DDL whenever a property is annotated with `@NotNull`.

**Comment by jo-elimu** [comment]
Possibly related: https://stackoverflow.com/a/51734560
